src/walk-manifest.js

```diff
@@ -114,7 +114,15 @@ var walkPlaylist = function(decrypt, basedir, uri, parent, manifestIndex) {
 		parent.content = new Buffer(parent.content.toString().replace(uri, path.relative(path.dirname(parent.file), manifest.file)));
 	}
 
-  manifest.content = syncRequest('GET', manifest.uri).getBody();
+  res = syncRequest('GET', manifest.uri);
+
+  // Don't throw an error if we get a non 200 just return an emtpy array.
```

Comment:
why is this needed? If an error is being returned couldn't this just be handled by a catch block in the calling code?

Reply:
That's definitely I guess that's why they handle it that way in `cli.js`. I'll revert it.

Reply:
so I think you were on to something here after all. The current throwing behavior will likely cause an all-or-nothing parse, which means that any bad URLs will lead to an exception which renders the rest of the results unusable (i.e. a bad URL in a rendition playlist will bubble up past the master playlist and any other renditions which may have had all valid URLs). This may or may not be relevant for calling code but I think in our case it is.

I'd advocate the ability to pass an error handler (which by default throws an error) which can be recursively passed through and therefore allow for better collection of errors throughout the tree as desired.

Reply:
@gkatsev @BrandonOCasey ^^ thoughts?

Reply:
Maybe the error handling can be done as part of the other refactoring PR?

Reply:
I think that would be the plan but just wanted to make sure that wasn't any immediate cause for alarm in the approach, thanks!

Reply:
yeah, from what I saw in that PR, if we are going with callbacks, I'd expected a node-style callback where errors are returned rather than thrown.